front/app/containers/Authentication/useSteps/stepConfig/signUpFlow.ts

```diff
@@ -202,10 +202,10 @@ export const signUpFlow = (
             return;
           }
 
-          if (requirements.special.group_membership === 'require') {
-            setCurrentStep('closed');
-            return;
-          }
+          // if (requirements.special.group_membership === 'require') {
```

Comment:
@luucvanderzee when I comment out this code, it works via the sign-up form.

I noticed that when I fill out the required field afterwards via the user menu > Complete profile, the current step was set to `'success'`. By commenting out this code, the same happens when I do so via the sign-up form. So it works now (the native survey button becomes available and I'm redirect to the survey without page refresh).

Questions I still have: 
1. why do we set current step to 'closed' when there's a group_membership required?
2. Is there a problem you can think of with showing the success step instead?

Reply:
Hey, I'm not completely sure what the bug is you're trying to solve, but  `requirements.special.group_membership === 'require'` actually means you are not allowed to participate, because you are not in the right smart group. So while yes, you filled out the custom fields successfully, your main objective (i.e. to participate) was not achieved successfully, because it turns out you're not allowed to participate. So the idea was to not show you the success step and instead close the modal, so that you could see the disabled button with a tooltip explaining what is happening.

Can you explain to me what you're trying to do exactly? Because if you encounter `requirements.special.group_membership === 'require'` in a situation where you should be allowed to participate, the bug is actually in the backend and not the frontend